pennylane/ops/functions/matrix.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the qml.matrix function.
+"""
+# pylint: disable=protected-access
+import pennylane as qml
+
+
+@qml.op_transform
+def matrix(op, *, wire_order=None):

```

Comment:
If this can take tapes etc, would you not call it `obj`?

Reply:
good point!

Reply:
Oh, because the wrapper makes sure that by the time it reaches the code in this function, it is an op?

Reply:
Yep exactly :) But the documentation will only show the docstring for the first one, hence why I've tried to keep all the information for op/tape/qfunc etc in a single docstring